src/sql/workbench/api/browser/mainThreadConnectionManagement.ts

```diff
@@ -146,8 +146,8 @@ export class MainThreadConnectionManagement extends Disposable implements MainTh
 			return undefined;
 		}
 
-		// Here we default to ConnectionType.editor which saves the connecton in the connection store by default
-		let connectionType = ConnectionType.editor;
+		// Here we default to ConnectionType.default which saves the connection in the connection store and server tree by default
+		let connectionType = ConnectionType.default;
```

Comment:
Lines 152 - 156 don't seem necessary with this change. Although I'm not sure offhand what the difference between "temporary" and "default" is for purposes of the connection dialog...did you check to see how those are treated differently and whether using temporary instead makes more sense?

Reply:
Temporary is also treated as a not to be saved profile..
https://github.com/microsoft/azuredatastudio/blob/e01e3b0e8ee25432b3cf64b021ef098d162e3785/src/sql/workbench/services/connection/browser/connectionDialogService.ts#L478-L481

Reply:
Ok, actually I guess that makes sense if "temporary" is truly supposed to be temporary - I wouldn't expect that to be saved to the server tree. Default it is!

(of course one of these days going and taking a hard look at how these connection types are used would be valuable I think. I'm suspicious of why we need something like this instead of just modifying the connection properties for the different contexts it's used in. The current behavior of having these random overrides is very much not intuitive or easy to understand as we're finding out...)